dashboard/modules/job/job_manager.py

```diff
@@ -137,12 +138,21 @@ def _exec_entrypoint(self, logs_path: str) -> subprocess.Popen:
                 terminated or killed upon user calling stop().
         """
         with open(logs_path, "w") as logs_file:
+            if self._runtime_env.get("pip") is not None:
+                # Replace "python" in the user's entrypoint script with the
+                # Python executable for the present runtime env.
+                cmd = f'python() {{ "{sys.executable}" "$@" ; }} && ' + self._entrypoint
+                executable = "/bin/bash"
```

Comment:
hmm why is this necessary?

Reply:
Otherwise an entrypoint script that uses `python` won't be able to import the pip packages, so the test `"python -c 'import pip_install_test'"` will fail.  (The driver scripts will also fail their imports outside of tasks and actors, it's the same as the Ray client server issue we discussed.)

The reason is that `pip` runtime_env actually changes the `python` executable that it uses for tasks and actors.

Reply:
Discussed offline, this is a hack that won't work in common scenarios like `rllib run`.  The underlying issue is specific to runtime envs, tracked here https://github.com/ray-project/ray/issues/22968.

Because that issue is independent of the one in this PR (fixing requirements files), I'll remove this hack from this PR so that just the part about `requirements.txt` can be merged independently.  After https://github.com/ray-project/ray/issues/22968 is fixed we can add back the `python -c "import pip-install-test"` unit test.